### PR TITLE
fix: allow beacon api req/res to use JSON

### PR DESCRIPTION
**Pull request by danyalprout**
### Description
* Adds a workaround for https://github.com/base-org/blob-archiver/issues/17.
* Updates dependencies (optimism/op-geth/go-eth-2 client)
* Removes some unessacary logging from the go-eth-2 client

Currently when using the go-eth-2 client to fetch blob sidecars it results in an error when dividing the response into sections [here](https://github.com/attestantio/go-eth2-client/blob/master/api/blobsidecars.go#L29).

### CODE REVIEW

File: common/beacon/client.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ func NewBeaconClient(ctx context.Context, cfg flags.BeaconConfig) (Client, error
 	cctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	c, err := http.New(cctx, http.WithAddress(cfg.BeaconURL), http.WithTimeout(cfg.BeaconClientTimeout))
+	c, err := http.New(cctx, http.WithAddress(cfg.BeaconURL), http.WithTimeout(cfg.BeaconClientTimeout), http.WithEnforceJSON(cfg.EnforceJSON), http.WithLogLevel(zerolog.ErrorLevel))
```
Comment (0x00101010) [inline common/beacon/client.go:23]: nit: should we make the log level configurable? is error level enough?

Comment (roberto-bayardo) [inline common/beacon/client.go:23]: why add WithLogLevel.ErrorLevel?

Comment (danyalprout) [reply common/beacon/client.go:23]: Going to address in @0x00101010's comment

Comment (danyalprout) [reply common/beacon/client.go:23]: The logs that this client outputs are mostly duplicated with the service logs (e.g. [here](https://github.com/base-org/blob-archiver/blob/master/archiver/service/archiver.go#L83-L112)).

It also don't hook into oplog and as such don't format nicely -- so I figured it was worth preferring the services logs over the libraries.

That being said -- I don't have a strong opinion here. Happy to keep them in if you prefer.

Comment (roberto-bayardo) [reply common/beacon/client.go:23]: I trust your judgement here. thanks!

**Review by roberto-bayardo — APPROVED** [review]

**Pull request merged**